dataRetriever/factory/containers/requestersContainer_test.go

```diff
@@ -0,0 +1,331 @@
+		assert.Equal(t, process.ErrWrongTypeInContainer, err)
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		c := containers.NewRequestersContainer()
+
+		key := "key"
+		val := &dataRetrieverTests.RequesterStub{}
+
+		_ = c.Add(key, val)
+		valRecovered, err := c.Get(key)
+
+		assert.True(t, val == valRecovered)
+		assert.Nil(t, err)
+	})
+}
+
+func TestRequestersContainer_Replace(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil value should error", func(t *testing.T) {
+		t.Parallel()
+
+		c := containers.NewRequestersContainer()
+
+		key := "key"
+		val := &dataRetrieverTests.RequesterStub{}
+
+		_ = c.Add(key, val)
+		err := c.Replace(key, nil)
+
+		valRecovered, _ := c.Get(key)
+
+		assert.Equal(t, process.ErrNilContainerElement, err)
+		assert.Equal(t, val, valRecovered)
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		c := containers.NewRequestersContainer()
+
+		key := "key"
+		val := &dataRetrieverTests.RequesterStub{}
+		val2 := &dataRetrieverTests.RequesterStub{}
+
+		_ = c.Add(key, val)
+		err := c.Replace(key, val2)
+
+		valRecovered, _ := c.Get(key)
+
+		assert.True(t, val2 == valRecovered)
+		assert.Nil(t, err)
+	})
+}
+
+func TestRequestersContainer_Remove(t *testing.T) {
+	t.Parallel()
+
+	c := containers.NewRequestersContainer()
+
+	key := "key"
+	val := &dataRetrieverTests.RequesterStub{}
+
+	_ = c.Add(key, val)
+	c.Remove(key)
+
+	valRecovered, err := c.Get(key)
+
+	assert.Nil(t, valRecovered)
+	assert.True(t, errors.Is(err, dataRetriever.ErrInvalidContainerKey))
+}
+
+func TestRequestersContainer_Len(t *testing.T) {
+	t.Parallel()
+
+	c := containers.NewRequestersContainer()
+
+	_ = c.Add("key1", &dataRetrieverTests.RequesterStub{})
+	assert.Equal(t, 1, c.Len())
+
+	_ = c.Add("key2", &dataRetrieverTests.RequesterStub{})
+	assert.Equal(t, 2, c.Len())
+
+	c.Remove("key1")
+	assert.Equal(t, 1, c.Len())
+}
+
+func TestRequestersContainer_RequesterKeys(t *testing.T) {
+	t.Parallel()
+
+	c := containers.NewRequestersContainer()
+
+	_ = c.Add("key1", &dataRetrieverTests.RequesterStub{})
+	_ = c.Add("key0", &dataRetrieverTests.RequesterStub{})
+	_ = c.Add("key2", &dataRetrieverTests.RequesterStub{})
+	_ = c.Add("a", &dataRetrieverTests.RequesterStub{})
+	c.Objects().Insert(1234, &dataRetrieverTests.RequesterStub{}) // key not string should be skipped
+
+	expectedString := "a, key0, key1, key2"
+
+	assert.Equal(t, expectedString, c.RequesterKeys())
+}
+
+func TestRequestersContainer_Iterate(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil handler should not panic", func(t *testing.T) {
+		t.Parallel()
+
+		defer func() {
+			r := recover()
+			if r != nil {
+				assert.Fail(t, "should not have panicked")
+			}
+		}()
+
+		c := containers.NewRequestersContainer()
+
+		_ = c.Add("key1", &dataRetrieverTests.RequesterStub{})
+		_ = c.Add("key2", &dataRetrieverTests.RequesterStub{})
+
+		c.Iterate(nil)
+	})
+	t.Run("invalid key should work", func(t *testing.T) {
+		t.Parallel()
+
```

Comment:
is the description correct? which is the invalid key?

Reply:
correct, removed this test